engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -172,12 +177,28 @@ public static AnalyzerContext createContext(
             // new columns shadow known aliases
             resultAlias.remove(sc.getName());
 
+            // execution dependencies are based on layer; the recomputation dependendencies are done by name (a barrier
+            // can reach across name aliases, but the formula inputs cannot)
+            final TIntArrayList execDeps = new TIntArrayList();
+            final Object[] respectedBarriers = sc.respectedBarriers();
+            if (respectedBarriers != null) {
+                for (final Object barrier : respectedBarriers) {
+                    final Integer layerForBarrier = barrierToLayerIndex.get(barrier);
+                    if (layerForBarrier == null) {
+                        throw new IllegalArgumentException(
```

Comment:
We _could_ ignore this, and then get rid of the barrier removal. I imagine a user might add respected barriers and then re-use a column. But maybe you hate that.

Reply:
I do hate that as we discussed.